aeon/schema/social.py

```diff
@@ -0,0 +1,88 @@
+"""Readers for data relevant to Social experiments."""
+
+import json
+from pathlib import Path
+
+import pandas as pd
+
+from aeon import util
+import aeon.io.reader as _reader
+
+
+class Pose(_reader.Harp):
+    """Reader for Harp-binarized tracking data given a model that outputs id, parts, and likelihoods."""
+
+    def __init__(self, pattern):
+        self.pattern = pattern
+        self.extension = "bin"
+
+    def read(self, file, ceph_proc_dir="/ceph/aeon/aeon/data/processed"):
+        """Reads data from the Harp-binarized tracking file."""
+        # Get config file from `file`, then bodyparts from config file.
+        model_dir = file.stem.replace("_", "/")
+        config_file_dir = Path(ceph_proc_dir + "/" + model_dir)
+        assert config_file_dir.exists(), f"Cannot find model dir {config_file_dir}"
+        config_file = get_config_file(config_file_dir)
+        parts = self.get_bodyparts(config_file)
+        # Using bodyparts, assign column names to Harp register values, and read data in default format.
+        columns = ["class", "class_likelihood"]
+        for part in parts:
+            columns.extend([f"{part}_x", f"{part}_y", f"{part}_likelihood"])
+        self.columns = columns
+        data = super().read(file)
+        # Set new columns, and reformat `data`.
+        new_columns = ["class", "class_likelihood", "part", "part_likelihood", "x", "y"]
+        new_data = pd.DataFrame(columns=new_columns)
+        part_data_list = [None] * len(parts)
```

Comment:
It is actually quicker to preallocate it this way, though essentially negligible in this case. "Avoid list appends like the plague" is generally a good rule to follow for code performance in python.

Reply:
I see, I guess this is because you cannot initialise a list with a starting capacity as you can in C#.

In that case I wonder if writing this as a list comprehension might be even faster? When I have this kind of code where lists are generated from other lists I tend to prefer comprehensions in general, sometimes defining aux functions if needed for clarity.

That should keep readability without compromising on performance. Maybe it is just my aesthetic bias that it doesn't feel right to me to initialise and manipulate list insertions like this.